src/quicktimevideo.cpp

```diff
@@ -0,0 +1,1622 @@
+  NextTrackID
+};
+enum trackHeaderTags {
+  TrackHeaderVersion,
+  TrackCreateDate,
+  TrackModifyDate,
+  TrackID,
+  TrackDuration = 5,
+  TrackLayer = 8,
+  TrackVolume,
+  ImageWidth = 19,
+  ImageHeight
+};
+enum mediaHeaderTags {
+  MediaHeaderVersion,
+  MediaCreateDate,
+  MediaModifyDate,
+  MediaTimeScale,
+  MediaDuration,
+  MediaLanguageCode
+};
+enum handlerTags { HandlerClass = 1, HandlerType, HandlerVendorID };
+enum videoHeaderTags { GraphicsMode = 2, OpColor };
+enum stream { Video, Audio, Hint, Null, GenMediaHeader };
+enum imageDescTags {
+  codec,
+  VendorID = 4,
+  SourceImageWidth_Height = 7,
+  XResolution,
+  YResolution,
+  CompressorName = 10,
+  BitDepth
+};
+enum audioDescTags { AudioFormat, AudioVendorID = 4, AudioChannels, AudioSampleRate = 7, MOV_AudioFormat = 13 };
+
+/*!
+  @brief Function used to check equality of a Tags with a
+      particular string (ignores case while comparing).
+  @param buf Data buffer that will contain Tag to compare
+  @param str char* Pointer to string
+  @return Returns true if the buffer value is equal to string.
+ */
+bool equalsQTimeTag(Exiv2::DataBuf& buf, const char* str) {
+  for (int i = 0; i < 4; ++i)
+    if (tolower(buf.data()[i]) != tolower(str[i]))
+      return false;
+  return true;
+}
+
+/*!
+  @brief Function used to ignore Tags and values stored in them,
+      since they are not necessary as metadata information
+  @param buf Data buffer that will contain Tag to compare
+  @return Returns true, if Tag is found in the ignoreList[]
+ */
+bool ignoreList(Exiv2::DataBuf& buf) {
+  const char ignoreList[13][5] = {
+      "mdat", "edts", "junk", "iods", "alis", "stsc", "stsz", "stco", "ctts", "stss", "skip", "wide", "cmvd",
+  };
+
+  for (int i = 0; i < 13; ++i)
+    if (equalsQTimeTag(buf, ignoreList[i]))
+      return true;
+
+  return false;
+}
+
+/*!
+  @brief Function used to ignore Tags, basically Tags which
+      contain other tags inside them, since they are not necessary
+      as metadata information
+  @param buf Data buffer that will contain Tag to compare
+  @return Returns true, if Tag is found in the ignoreList[]
+ */
+bool dataIgnoreList(Exiv2::DataBuf& buf) {
+  const char ignoreList[8][5] = {
+      "moov", "mdia", "minf", "dinf", "alis", "stbl", "cmov", "meta",
+  };
+
+  for (int i = 0; i < 8; ++i)
+    if (equalsQTimeTag(buf, ignoreList[i]))
+      return true;
+
+  return false;
+}
+
+/*!
+  @brief Function used to convert buffer data into 64-bit
+      signed integer, information stored in Big Endian format
+  @param buf Data buffer that will contain data to be converted
+  @return Returns a signed 64-bit integer
+ */
+int64_t returnBufValue(Exiv2::DataBuf& buf, int n = 4) {
+  int64_t temp = 0;
+  for (int i = n - 1; i >= 0; i--)
+#ifdef _MSC_VER
+    temp = temp + static_cast<int64_t>(buf.data()[i] * (pow(static_cast<float>(256), n - i - 1)));
+#else
+    temp = temp + buf.data()[i] * (pow((float)256, n - i - 1));
+#endif
+
+  return temp;
+}
```

Comment:
This looks like a weird byte-swapping implementation. I think we can just use `DataBuf::read_uint32()` and friends instead.